mscore/scoreview.cpp

```diff
@@ -5540,14 +5546,96 @@ void ScoreView::cmdAddChordName()
       harmony->setTrack(cr->track());
       harmony->setParent(cr->segment());
       _score->undoAddElement(harmony);
-
       _score->select(harmony, SelectType::SINGLE, 0);
       startEdit(harmony);
       _score->setLayoutAll();
       _score->update();
       }
 
 //---------------------------------------------------------
+//   cmdAddAnnotation
+//---------------------------------------------------------
+
+void ScoreView::cmdAddAnnotation()
+      {
+      if (!_score->checkHasMeasures())
+            return;
+      if (noteEntryMode())          // force out of entry mode
+            sm->postEvent(new CommandEvent("note-input"));
+
+      _score->startCmd();
+        ChordRest* cr = _score->getSelectedChordRest();
+      if (!cr)
+            return;
+
+      TextAnnotation* t = new TextAnnotation(_score);
+      t->setTrack(cr->track());
+      t->setTextStyleType(TextStyleType::ANNOTATION);
+      t->setParent(cr->segment());
+      t->setColor(Qt::red);
+      if (t) {
+            _score->undoAddElement(t);
+            _score->select(t, SelectType::SINGLE, 0);
+            _score->endCmd();
+            startEdit(t);
+            }
+      else
+            _score->endCmd();
+
+      }
+
+//---------------------------------------------------------
+//   cmdAddRangeAnnotation
+//---------------------------------------------------------
+
+void ScoreView::cmdAddRangeAnnotation()
+      {
+      if (!_score->selection().isRange()) {
+            ChordRest* cr = _score->getSelectedChordRest();
+            if (!cr)
+                  return;
+            int stick = cr->tick();
+            int etick = stick + cr->actualTicks();
+            int sstaff = cr->staffIdx();
+            int strack = sstaff * VOICES;
+            RangeAnnotation* rangeAnn = new RangeAnnotation(_score);
+            rangeAnn->setParent(0);
+            rangeAnn->setTick(stick);
+            rangeAnn->setTick2(etick);
+            rangeAnn->setTrack(strack);
+            rangeAnn->setStaffStart(sstaff);
+            rangeAnn->setStaffEnd(sstaff);
+            QColor yellow(255, 255, 0, 127);
+            rangeAnn->setColor(yellow);
```

Comment:
ultimately this (and other properties) should probably be style settings, but this is fine for now